webcompat/views.py

```diff
@@ -302,13 +302,18 @@ def cssfixme():
 @app.errorhandler(500)
 def custom_error_handler(err):
     # log the exception stack trace
-    app.logger.exception("!!!")
-    if api_call(request):
-        return api_message(err.code)
-    return render_template(
-        'error.html',
-        error_code=err.code,
-        error_message=ERROR_DICT[err.code]), err.code
+    # (but don't bother for localhost because the
+    # Flask debugger is already enabled)
+    if not app.config['LOCALHOST']:
+        app.logger.exception("Exception thrown:")
+    try:
+        if api_call(request):
+            return api_message(err.code)
+        return render_template('error.html', error_code=err.code,
+                               error_message=ERROR_DICT[err.code]), err.code
+    except AttributeError:
+        # Somethign bad happened, we're not dealing with an HTTPError
+        abort(500)
```

Comment:
> And this seems recursive. abort(500) will call the custom error handler. Basically called twice?

I tested this by adding a `print('custom_error_handler')` at the top of `custom_error_handler`:

```
127.0.0.1 - - [02/May/2016 12:43:24] "GET / HTTP/1.1" 500 -
Error on request:
Traceback (most recent call last):
  File "/Users/miket/dev/compat/webcompat.com/env/lib/python2.7/site-packages/werkzeug/serving.py", line 180, in run_wsgi
    execute(self.server.app)
  File "/Users/miket/dev/compat/webcompat.com/env/lib/python2.7/site-packages/werkzeug/serving.py", line 168, in execute
    application_iter = app(environ, start_response)
  File "/Users/miket/dev/compat/webcompat.com/env/lib/python2.7/site-packages/flask/app.py", line 1836, in __call__
    return self.wsgi_app(environ, start_response)
  File "/Users/miket/dev/compat/webcompat.com/env/lib/python2.7/site-packages/flask/app.py", line 1820, in wsgi_app
    response = self.make_response(self.handle_exception(e))
  File "/Users/miket/dev/compat/webcompat.com/env/lib/python2.7/site-packages/flask/app.py", line 1410, in handle_exception
    return handler(e)
  File "/Users/miket/dev/compat/webcompat.com/webcompat/views.py", line 317, in custom_error_handler
    abort(500)
  File "/Users/miket/dev/compat/webcompat.com/env/lib/python2.7/site-packages/werkzeug/exceptions.py", line 605, in __call__
    raise self.mapping[code](*args, **kwargs)
InternalServerError: 500: Internal Server Error
```
If I actually call `custom_error_hander(500)`, rather than `abort(500)` I get the recursion error:

```
[snip a billion lines]
    custom_error_handler(500)
  File "/Users/miket/dev/compat/webcompat.com/webcompat/views.py", line 317, in custom_error_handler
    custom_error_handler(500)
  File "/Users/miket/dev/compat/webcompat.com/webcompat/views.py", line 317, in custom_error_handler
    custom_error_handler(500)
  File "/Users/miket/dev/compat/webcompat.com/webcompat/views.py", line 317, in custom_error_handler
    custom_error_handler(500)
  File "/Users/miket/dev/compat/webcompat.com/webcompat/views.py", line 317, in custom_error_handler
    custom_error_handler(500)
  File "/Users/miket/dev/compat/webcompat.com/webcompat/views.py", line 317, in custom_error_handler
    custom_error_handler(500)
  File "/Users/miket/dev/compat/webcompat.com/webcompat/views.py", line 317, in custom_error_handler
    custom_error_handler(500)
  File "/Users/miket/dev/compat/webcompat.com/webcompat/views.py", line 317, in custom_error_handler
    custom_error_handler(500)
  File "/Users/miket/dev/compat/webcompat.com/webcompat/views.py", line 311, in custom_error_handler
    if api_call(request):
  File "/Users/miket/dev/compat/webcompat.com/webcompat/views.py", line 322, in api_call
    if (request.path.startswith('/api/') and
  File "/Users/miket/dev/compat/webcompat.com/env/lib/python2.7/site-packages/werkzeug/local.py", line 338, in __getattr__
    return getattr(self._get_current_object(), name)
  File "/Users/miket/dev/compat/webcompat.com/env/lib/python2.7/site-packages/werkzeug/local.py", line 297, in _get_current_object
    return self.__local()
  File "/Users/miket/dev/compat/webcompat.com/env/lib/python2.7/site-packages/flask/globals.py", line 18, in _lookup_req_object
    top = _request_ctx_stack.top
  File "/Users/miket/dev/compat/webcompat.com/env/lib/python2.7/site-packages/werkzeug/local.py", line 163, in top
    return self._local.stack[-1]
  File "/Users/miket/dev/compat/webcompat.com/env/lib/python2.7/site-packages/werkzeug/local.py", line 69, in __getattr__
    return self.__storage__[self.__ident_func__()][name]
RuntimeError: maximum recursion depth exceeded
```